test/api/endpoints/test_ogc.py

```diff
@@ -0,0 +1,765 @@
+                headers['Content-Type'] = 'application/json'
+                
+            if method.upper() == 'GET':
+                return self.client.get(url, headers=headers)
+            elif method.upper() == 'POST':
+                return self.client.post(url, data=json.dumps(data) if data is not None else None, headers=headers)
+            elif method.upper() == 'PUT':
+                return self.client.put(url, data=json.dumps(data) if data is not None else None, headers=headers)
+            elif method.upper() == 'DELETE':
+                return self.client.delete(url, headers=headers)
+
+    def test_processes_get_returns_deployed_processes(self):
+        """Test: GET /ogc/processes returns list of deployed processes"""
+        with app.app_context():
+            # Given a deployed process
+            member = self._create_test_member()
+            process = self._create_test_process(member)
+            
+            # When requesting all processes
+            response = self.client.get('/api/ogc/processes')
+            
+            # Then deployed processes should be returned
+            self.assertEqual(response.status_code, 200)
+            data = response.get_json()
+            self.assertIn('processes', data)
+            self.assertEqual(len(data['processes']), 1)
+            self.assertEqual(data['processes'][0]['id'], 'test-process')
+            self.assertEqual(data['processes'][0]['title'], 'Test Process')
+            self.assertEqual(data['processes'][0]['author'], 'testuser')
+
+    def test_processes_get_returns_empty_when_no_deployed_processes(self):
+        """Test: GET /ogc/processes returns empty list when no deployed processes"""
+        # When requesting all processes with none deployed
+        response = self.client.get('/api/ogc/processes')
+        
+        # Then empty process list should be returned
+        self.assertEqual(response.status_code, 200)
+        data = response.get_json()
+        self.assertIn('processes', data)
+        self.assertEqual(len(data['processes']), 0)
+
+    @patch('api.auth.security.get_authorized_user')
+    def test_processes_post_requires_authentication(self, mock_get_user):
+        """Test: POST /ogc/processes requires authentication"""
+        with app.app_context():
+            # Given no authenticated user
+            mock_get_user.return_value = None
+            
+            # When attempting to post a process without authentication
+            process_data = {
+                "executionUnit": {
+                    "href": "https://example.com/test.cwl"
+                }
+            }
+            
+            response = self.client.post('/api/ogc/processes',
+                                      data=json.dumps(process_data),
+                                      content_type='application/json')
+            
+            # Then authentication should be required
+            self.assertEqual(response.status_code, 403)
```

Comment:
For invalid authentication use 401 as return code. 403 is used for unauthorized access.